.github/workflows/pull_request.yml

```diff
@@ -134,11 +180,15 @@ jobs:
     - name: check perl
       run: make check_perl
 
-  tests:
-    name: 🐪 Perl unit tests
+  unit_tests:
+    name: 🐪 Perl Unit Tests
     needs: [filter, build_backend]
     runs-on: ubuntu-latest
-    if: ((github.event_name == 'push' && github.ref == 'refs/heads/main') || github.event_name == 'pull_request') && needs.filter.outputs.code_modified == 'true'
+    if: github.event_name == 'pull_request' && needs.filter.outputs.code_modified == 'true'
```

Comment:
We need unit tests and integration tests to run on the `main` branch to keep code coverage in Codecov up-to-date, see #11986